web3.js/src/connection.ts

```diff
@@ -297,6 +301,17 @@ export type BlockheightBasedTransactionConfirmationStrategy = {
   signature: TransactionSignature;
 } & BlockhashWithExpiryBlockHeight;
 
+/**
+ * @internal
+ */
+function createRPCError(
+  error: Readonly<{code: unknown; message: string; data?: any}>,
+  customMessage: string,
+): SolanaJSONRPCError {
+  const code = coerceSolanaJSONRPCErrorCode(error.code);
+  return new SolanaJSONRPCError(`${customMessage}: ${error.message}`, code);
```

Comment:
One unfortunate side-effect of _constructing_ the error here is that all of the _stack traces_ will point here. The real error site will be one line below this line in the stacktrace.

Reply:
Can we construct the error where we throw it then? Concatenating the custom and error message can happen in the `SolanaJSONRPCError` constructor

Reply:
Yeah; if there's no need for the extra step of coercion, then I don't need this wrapper.

Reply:
Ugh. Do you know why `code` is spec'd as `unknown()` here?

https://github.com/solana-labs/solana/blob/master/web3.js/src/connection.ts#L337